shared/src/address/mod.rs

```diff
@@ -374,7 +374,13 @@ mod tests {
 
 /// Checksum calculates the 4 byte checksum hash
 pub fn checksum(ingest: &[u8]) -> Vec<u8> {
-    blake2b_variable(ingest, CHECKSUM_HASH_LEN)
+    blake2b_simd::Params::new()
```

Comment:
What was the reason to remove this helper function?

Reply:
I wanted to make the hasher dependency more explicit, and not just randomly being pulled in. If we want to keep the helpers I would suggest they should be their own crate

Reply:
But this doesn't really change anything as this crate still depends on blake2b, right?

Reply:
Right now, it doesn't change anything for most crates, but the more this gets split apart the more useful it becomes. Can also put it back in here 🤷 for now, but don't want to put it into the encoding crate.